torch/cuda/amp/common.py

```diff
@@ -0,0 +1,7 @@
+import os
+import torch
+
+
+def cuda_or_xla_gpu_available():
+    # use `XRT_DEVICE_MAP` to check whether GPU is enabled in torch/xla
+    return torch.cuda.is_available() or "GPU" in os.getenv("XRT_DEVICE_MAP", "")
```

Comment:
This is horrible and I hate it (and it also is probably slow) but I'm not sure what a good replacement for it is.

Reply:
(The other reason I hate it is it's not clear at all why `XRT_DEVICE_MAP` is the *exclusive* mechanism by which GPU support can be turned on in XLA; if it is non-exclusive, then this implementation is wrong)

Reply:
Haha, I also don't like this.

Do you think the implementation below is acceptable?
```
from importlib.util import find_spec

def cuda_or_xla_gpu_available():
	return torch.cuda.is_available or find_spec('torch_xla')
```

Reply:
I think this is the germ of a good idea.

As directly cited here, I still don't think this is right, because simply *having* `torch_xla` installed will cause this to advertise that XLA GPU is available (even if it isn't). But that made me wonder "why do we have this function at all" and it is really only used in two spots, basically to give a helpful error message when people are trying to use AMP in situations where it couldn't possibly help.

This suggests to me that it is OK for this to be somewhat heuristic. The concept is that you don't need to tell exactly when CUDA/XLA GPU is available; you just have to be able to tell when they are *definitely not available*. So literally negate the body of this function, rename it to something like `amp_definitely_not_available()` (not a great name, feel free to pick your own), and I'm good with this implementation.

Reply:
Use `amp_definitely_not_available` is more reasonable. And it is done.